pandas/core/dtypes/concat.py

```diff
@@ -45,6 +46,8 @@ def get_dtype_kinds(l):
             # if to_concat contains different tz,
             # the result must be object dtype
             typ = str(arr.dtype)
+        elif is_range(arr):
```

Comment:
you can simply check ``isinstance(arr, ABCRangeIndex)``, you are special casing this so I don't find this a problem. We don't have a special 'type' for this index so ``is_int64_dtype`` would not work here.